### PR TITLE
Sighting quicksearch UI

### PR DESCRIPTION
**Pull request by brmscheiner**

### CODE REVIEW

File: src/models/sighting/useSightingTermQuery.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import useFetch from '../../hooks/useFetch';
+import { getSightingTermQueryKey } from '../../constants/queryKeys';
+
+export default function useSightingTermQuery(searchTerm) {
+  const query = {
+    query_string: {
+      query: `*${searchTerm}*`,
+      fields: ['guid'],
```
Comment (brmscheiner) [inline src/models/sighting/useSightingTermQuery.js:8]: we will add other fields to search on when they come out

**Review by Emily-Ke — APPROVED** [review]
One minor comment.

File: src/components/AuthenticatedAppHeader/IndividualsButton.jsx [hunk 2]
```diff
@@ -57,159 +28,70 @@ export default function IndividualsButton() {
...
-                    <Text variant="caption">{`Created on ${createdDate}`}</Text>
-                  }
+              return (
+                <SearchResult
+                  key={individualGuid}
+                  avatarLetter={avatarLetter}
+                  href={`/individuals/${individualGuid}`}
+                  onClick={closePopover}
+                  primaryText={displayString}
+                  secondaryText={`Created on ${createdDate}`}
```
Comment (Atticus29) [inline src/components/AuthenticatedAppHeader/IndividualsButton.jsx:88]: i18n

File: src/components/AuthenticatedAppHeader/SearchButton.jsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+  useOnEnter(() => {
+    if (inputValue !== '') onSearch(inputValue);
+  });
+
+  const closePopover = () => {
+    setAnchorEl(null);
+  };
+
+  const open = Boolean(anchorEl);
```
[the remark this answers is not in the document]
Comment (Atticus29) [reply src/components/AuthenticatedAppHeader/SearchButton.jsx:42]: Or at least this pattern caused a bug like that for the notification popover.

File: src/components/AuthenticatedAppHeader/SearchResult.jsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        <Avatar>{avatarLetter}</Avatar>
+      </ListItemAvatar>
+
+      <ListItemText
+        primary={
+          <Link onClick={onClick} href={href} noUnderline>
+            {primaryText}
+          </Link>
+        }
+        secondary={<Text variant="caption">{secondaryText}</Text>}
```
Comment (Atticus29) [inline src/components/AuthenticatedAppHeader/SearchResult.jsx:30]: will there always be secondaryText?

File: src/components/AuthenticatedAppHeader/SightingsButton.jsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  const [searchTerm, setSearchTerm] = useState('');
+
+  const {
+    data: searchResults,
+    loading,
+    error,
+  } = useSightingTermQuery(searchTerm);
+
+  const resultsCurrent = inputContent === searchTerm;
+  const noResults = searchResults && searchResults.length === 0;
```
Comment (Atticus29) [inline src/components/AuthenticatedAppHeader/SightingsButton.jsx:22]: different from searchResults?.length === 0
?

File: src/components/AuthenticatedAppHeader/SightingsButton.jsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  const mappableSearchResults = resultsCurrent
+    ? searchResults || []
```
Comment (Atticus29) [inline src/components/AuthenticatedAppHeader/SightingsButton.jsx:24]: I think parentheses here to disambiguate would be helpful.

**Review by Atticus29 — COMMENTED** [review]
A few little things, but great job!

File: src/components/AuthenticatedAppHeader/IndividualsButton.jsx [hunk 7]
```diff
@@ -57,159 +28,70 @@ export default function IndividualsButton() {
...
+                individual,
+                'AdoptionName',
+              );
+              const defaultName = deriveIndividualName(
+                individual,
+                'FirstName',
+                'Unnamed Individual',
+              );
+              const displayString = adoptionName
+                ? `${defaultName} (${adoptionName})`
```
[the remark this answers is not in the document]
Comment (brmscheiner) [reply src/components/AuthenticatedAppHeader/IndividualsButton.jsx:72]: it's called a ternary!

File: src/components/AuthenticatedAppHeader/SearchButton.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (brmscheiner) [reply src/components/AuthenticatedAppHeader/SearchButton.jsx:42]: can you explain more about the bug? it seems to me like when anchorEl becomes null the popover will close here. what code change is needed?

File: src/components/AuthenticatedAppHeader/SightingsButton.jsx (hunk 5 above)
Comment (brmscheiner) [reply src/components/AuthenticatedAppHeader/SightingsButton.jsx:22]: image
this was the case i was thinking about.. though i suppose the result is the same so maybe i can switch it!

File: src/components/AuthenticatedAppHeader/SightingsButton.jsx (hunk 6 above)
Comment (brmscheiner) [reply src/components/AuthenticatedAppHeader/SightingsButton.jsx:24]: the formatter decides on the parentheses =/

File: src/components/AuthenticatedAppHeader/SearchResult.jsx (hunk 4 above)
Comment (brmscheiner) [reply src/components/AuthenticatedAppHeader/SearchResult.jsx:30]: everywhere its currently used yes.

File: src/components/AuthenticatedAppHeader/SearchButton.jsx [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+  const id = open ? 'simple-popover' : undefined;
```
[the remark this answers is not in the document]
Comment (brmscheiner) [reply src/components/AuthenticatedAppHeader/SearchButton.jsx:43]: this is an html id, we dont need to translate these

File: src/components/AuthenticatedAppHeader/IndividualsButton.jsx [hunk 9]
```diff
@@ -57,159 +28,70 @@ export default function IndividualsButton() {
...
+      inputValue={inputContent}
+      onChangeInputValue={setInputContent}
+      onSearch={setSearchTerm}
+      instructionsLabelId="SEARCH_INDIVIDUALS_INSTRUCTION"
+      linkLabelId="EXPLORE_INDIVIDUALS"
+      linkHref="/individuals"
+    >
+      {closePopover => (
+        <>
+          {showDivider ? <Divider /> : null}
```
[the remark this answers is not in the document]
Comment (brmscheiner) [reply src/components/AuthenticatedAppHeader/IndividualsButton.jsx:43]: correct

File: src/components/AuthenticatedAppHeader/SearchButton.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (brmscheiner) [reply src/components/AuthenticatedAppHeader/SearchButton.jsx:42]: from DMs - tested for the bug and it's not there

File: src/components/AuthenticatedAppHeader/IndividualsButton.jsx [hunk 10]
```diff
@@ -38,178 +23,83 @@ export default function IndividualsButton() {
...
-  const handleClose = () => {
-    setAnchorEl(null);
-  };
-
-  const open = Boolean(anchorEl);
-  const id = open ? 'simple-popover' : undefined;
-
   const resultsCurrent = inputContent === searchTerm;
-  const noResults = searchResults && searchResults.length === 0;
+  const noResults = searchResults?.length === 0;
```
Comment (Emily-Ke) [inline src/components/AuthenticatedAppHeader/IndividualsButton.jsx:27]: If searchResults is nullish, noResults will be false. Don't you want it to be true in this case?

Comment (brmscheiner) [reply src/components/AuthenticatedAppHeader/IndividualsButton.jsx:27]: i only want it to be true if searchResults are available and there are none. i want false if it's loading or in an error state

**Review by Atticus29 — APPROVED** [review]
LGTM!

**Pull request merged**